plugins/kotlin/analysis/src/org/jetbrains/kotlin/idea/caches/resolve/IDEKotlinAsJavaSupport.kt

```diff
@@ -331,22 +332,30 @@ open class IDEKotlinAsJavaSupport(private val project: Project) : KotlinAsJavaSu
         val javaFileStub = ClsJavaStubByVirtualFileCache.getInstance(project).get(classFile) ?: return null
         javaFileStub.psiFactory = ClsWrapperStubPsiFactory.INSTANCE
         val manager = PsiManager.getInstance(mirrorFile.project)
-        val fakeFile = object : ClsFileImpl(ClassFileViewProvider(manager, classFile)) {
-            override fun getNavigationElement(): PsiElement {
-                if (correspondingClassOrObject != null) {
-                    return correspondingClassOrObject.navigationElement.containingFile
-                }
-                return super.getNavigationElement()
+        val fakeFile = FakeClsFileImpl(manager, classFile, mirrorFile, correspondingClassOrObject, javaFileStub)
```

Comment:
Could you please explain why this change in FE1.0 Plugin is needed?

Reply:
Ugh... not quite necessary. Just doing the same refactoring while doing so in FIR LC.